src/components/shared/epic.tsx

```diff
@@ -0,0 +1,124 @@
+// ----- Imports ----- //
+import { css, SerializedStyles } from '@emotion/core';
+import { basePx, headlineFont } from 'styles';
+import { from } from '@guardian/src-foundations/mq';
+import { palette, brandAlt } from '@guardian/src-foundations';
+import React, { useState, useEffect, useRef } from 'react';
+import { nativeClient } from 'native/nativeApi';
+import { SvgArrowRightStraight } from "@guardian/src-svgs"
+import { ThemeProvider } from 'emotion-theming'
+import { Button, buttonReaderRevenue } from '@guardian/src-button'
+
+
+// ----- Styles ----- //
+
+const EpicStyles = (): SerializedStyles => css`
+        width: calc(100% - ${basePx(4)});
+        margin: ${basePx(1)};
+
+        ${from.wide} {
+            margin: ${basePx(1, 0)};
+        }
+
+        clear: both;
+
+        border-top: 1px solid ${brandAlt[400]};
+        background: ${palette.neutral[97]};
+        padding: ${basePx(1)};
+        font-family: 'Guardian Text Egyptian Web';
+        clear: left;
+
+        h1:first-of-type {
+            ${headlineFont}
+        }
+
+        button {
+            margin: ${basePx(0, 1, 1, 0)};
+        }
+
+        .button-container {
+            margin-top: 3rem;
+        }
+
+        svg {
+            margin-left: 8px;
+            margin-top: -4px;
+            vertical-align: middle;
+        }
+
+        mark {
+            background: ${brandAlt[400]};
+            padding: .1rem .125rem;
+        }
+`;
+
+
+// ----- Component ----- //
+
+interface EpicProps {
+    title: string;
+    body: string;
+    firstButton: string;
+    secondButton: string | undefined;
+}
+
+const isElementPartiallyInViewport = (el: React.MutableRefObject<HTMLDivElement>): boolean => {
```

Comment:
Any thoughts on [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)?

Reply:
Hmm yeah we could upgrade this to use that... we're using `MutationObserver` elsewhere.  Although I guess it wouldn't work for iOS 12 users...

Reply:
Safari 12.1 [shipped with iOS 12.2](https://developer.apple.com/documentation/safari_release_notes/safari_12_1_release_notes), so I guess anyone on iOS prior to 12.2.